### PR TITLE
Operator doc

### PR DESCRIPTION
**Pull request by mramendi**

### CODE REVIEW

File: doc-content/enterprise-only/openshift/registry-authentication-proc.adoc [hunk 1]
```diff
@@ -0,0 +1,24 @@
+[id='registry-authentication-proc']
+= Ensuring your environment is authenticated to the Red Hat registry
```
Comment (tchughesiv) [inline doc-content/enterprise-only/openshift/registry-authentication-proc.adoc]: This section is optional. It's only necessary for clusters that don't already have the rhpam/rhdm 7.3 imagestreams available in the `openshift` project space.

File: doc-content/enterprise-only/openshift/operator-createapp-proc.adoc [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+*** `<url>`: The URL for the Git repository that contains the source for your decision service.
+*** `<branch>`: The branch in the Git repository.
+*** `<directory>`: The path to the source within the project downloaded from the Git repository (??? where is the artifact directory ???)
++
+** The following snippet sets the number of {KIE_SERVERS} that are managed by {CENTRAL} or {CENTRAL} Monitoring in your environment: 
++
+[subs="attributes,verbatim,macros"]
+----
+  objects:
+    server:
```
Comment (tchughesiv) [inline doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: We've consolidated the server/servers section into a single `servers` list. It's more intuitive and opens door for additional functionality. See -
https://github.com/kiegroup/kie-cloud-operator/blob/1.0.x/deploy/examples/server_config.yaml
https://github.com/kiegroup/kie-cloud-operator/blob/1.0.x/deploy/examples/immutable_s2i.yaml

File: doc-content/enterprise-only/openshift/operator-createapp-proc.adoc [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
++
+** The following snippet sets up RH-SSO authentication. You must ensure that the realm and clients exist in your RH-SSO deployment. The number of listed servers must be the same as the number of actual {KIE_SERVERS} in your environment. (??? what about the option where deployment auto-creates the clients - do operators support this?)
++
+[subs="attributes,verbatim,macros"]
+----
+  auth:
+    sso:
+      url: https://rh-sso.example.com:8443
+      realm: rhpam
+      clients:
```
Comment (tchughesiv) [inline doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: This has all been moved into the individual object sections... see -
https://github.com/kiegroup/kie-cloud-operator/blob/1.0.x/deploy/examples/auth_sso.yaml

File: assemblies/assembly_openshift-operator/main.adoc [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+IMPORTANT: ReadWriteMany mode is not supported on OpenShift Online and OpenShift Dedicated.
+
+include::{enterprise-dir}/openshift/ba-openshift-overview-con.adoc[leveloffset=+1]
+include::{enterprise-dir}/openshift/dm-openshift-prepare-con.adoc[leveloffset=+1]
+include::{enterprise-dir}/openshift/registry-authentication-proc.adoc[leveloffset=+2]
+include::{enterprise-dir}/openshift/secrets-kie-create-proc.adoc[leveloffset=+2]
+include::{enterprise-dir}/openshift/secrets-central-create-proc.adoc[leveloffset=+2]
+include::{enterprise-dir}/openshift/secrets-smartrouter-create-proc.adoc[leveloffset=+2]
+include::{enterprise-dir}/openshift/externaldb-build-proc.adoc[leveloffset=+2]
+include::{enterprise-dir}/openshift/glusterfs-reconfig-proc.adoc[leveloffset=+2]
```
Comment (tchughesiv) [inline assemblies/assembly_openshift-operator/main.adoc]: what's the purpose of the glusterfs instructions?

File: assemblies/assembly_openshift-operator/main.adoc [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+* If you intend to scale any of the {CENTRAL} or {CENTRAL} Monitoring pods, your OpenShift environment must support persistent volumes with ReadWriteMany mode.
++
+IMPORTANT: ReadWriteMany mode is not supported on OpenShift Online and OpenShift Dedicated.
+
+include::{enterprise-dir}/openshift/ba-openshift-overview-con.adoc[leveloffset=+1]
+include::{enterprise-dir}/openshift/dm-openshift-prepare-con.adoc[leveloffset=+1]
+include::{enterprise-dir}/openshift/registry-authentication-proc.adoc[leveloffset=+2]
+include::{enterprise-dir}/openshift/secrets-kie-create-proc.adoc[leveloffset=+2]
+include::{enterprise-dir}/openshift/secrets-central-create-proc.adoc[leveloffset=+2]
+include::{enterprise-dir}/openshift/secrets-smartrouter-create-proc.adoc[leveloffset=+2]
```
Comment (tchughesiv) [inline assemblies/assembly_openshift-operator/main.adoc]: do we need any of the above secrets includes with the operator? we don't require them, but its certainly an option for users.

Comment (mramendi) [reply assemblies/assembly_openshift-operator/main.adoc]: Secrets are needed as long as they are required for production

File: assemblies/assembly_openshift-operator/main.adoc (hunk 4 above)
Comment (mramendi) [reply assemblies/assembly_openshift-operator/main.adoc]: I was told previously that this is needed for any RHPAM deployment

File: doc-content/enterprise-only/openshift/operator-createapp-proc.adoc [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+            reference: <branch>
+            contextDir: <directory>
+----
++
+Replace the following values:
++
+*** `<deployment>`: The identifying information of the decision service (KJAR file) that is built from your source. The format is: `<containerId>=<groupId>:<artifactId>:<version>`. You can provide two or more KJAR files using the `|` separator, for example: `containerId=groupId:artifactId:version|c2=g2:a2:v2`. The Maven build process must produce all these files from the source in the Git repository.
+*** `<url>`: The URL for the Git repository that contains the source for your decision service.
+*** `<branch>`: The branch in the Git repository.
+*** `<directory>`: The path to the source within the project downloaded from the Git repository (??? where is the artifact directory ???)
```
Comment (tchughesiv) [inline doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: @ruromero please answer misha's question about contextDir

Comment (mramendi) [reply doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: I will remove the question from the doc now, should have done it before, thanks.. I will create a JIRA about artefact dir, which is distinct from context dir, 

Are there any more comments for this doc @tchughesiv ?

Comment (tchughesiv) [reply doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: no, LGTM!

Comment (ruromero) [reply doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: the `contextDir` is what defines the directory where to take the source from for the build the name is the same in the templates.

File: doc-content/enterprise-only/openshift/operator-createapp-proc.adoc [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+. Enter your project in the OpenShift Web cluster console. 
+. In the OpenShift Web console navigation panel, select *Operators* and then *Subscriptions*.
+. Click the name of the subscription that containt `businessautomation`. Information about this subscription is displayed.
+. Under the *Installed version* heading, click the version name of the subscription. An overview of the subscription is displayed.
+. Click the *Create KieApp* button. A YAML source is displayed.
+. Set the *name* field to an application name that is unique in the project.
+. Set the *environment* field to the required environment type. Each of the types is a default deployment pattern. You can modify the pattern by editing this YAML source; you can also modify the deployment after it is completed. The following types are available:
+
+** `rhpam-trial`: Includes {CENTRAL} and a {KIE_SERVER}. You can set it up quickly and use it to evaluate or demonstrate developing and running assets. However, the environment does not use any persistent storage, and any work you do in the environment is not saved.
```
Comment (sterobin) [inline doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: Maybe for consistency with the others: "A trial environment for ... . This environment includes {CENTRAL} and ..."

File: doc-content/enterprise-only/openshift/operator-createapp-proc.adoc [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+** `rhpam-production-immutable`: An alternate environment for running existing services for staging and production purposes. This environment includes {CENTRAL} Monitoring and Smart Router; you can configure one or more {KIE_SERVER} replicated pods that build a service from source. In this environment, when you deploy a {KIE_SERVER} pod, it builds an image that loads and starts a service or group of services. You cannot stop any service on the pod or add any new service to the pod. If you want to use another version of a service or modify the configuration in any other way, you deploy a new server image and displace the old one. In this system, the {KIE_SERVER} runs like any other pod on the OpenShift environment; you can use any container-based integration workflows and do not need to use any other tools to manage the pods.
```
Comment (sterobin) [inline doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: "Smart Router. You can configure" (no semicolon imo)
"environment. You can use any" (ditto)

File: doc-content/enterprise-only/openshift/operator-createapp-proc.adoc [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+. Click the *Create KieApp* button. A YAML source is displayed.
+. Set the *name* field to an application name that is unique in the project.
+. Set the *environment* field to the required environment type. Each of the types is a default deployment pattern. You can modify the pattern by editing this YAML source; you can also modify the deployment after it is completed. The following types are available:
+
+** `rhpam-trial`: Includes {CENTRAL} and a {KIE_SERVER}. You can set it up quickly and use it to evaluate or demonstrate developing and running assets. However, the environment does not use any persistent storage, and any work you do in the environment is not saved.
+** `rhpam-production`: An environment for running existing services for staging and production purposes. This environment includes {CENTRAL} Monitoring, Smart Router, and two groups of {KIE_SERVER} pods; you can deploy and undeploy services on every such group and also scale the group up or down as necessary. Use {CENTRAL} Monitoring to deploy, run, and stop the services and to monitor their execution.
```
Comment (sterobin) [inline doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: "pods. You can deploy" (no semicolon imo)

File: doc-content/enterprise-only/openshift/operator-createapp-proc.adoc [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+          kieServerContainerDeployment: <deployment>
+          gitSource:
+            uri: <url>
+            reference: <branch>
+            contextDir: <directory>
+----
++
+Replace the following values:
++
+*** `<deployment>`: The identifying information of the decision service (KJAR file) that is built from your source. The format is: `<containerId>=<groupId>:<artifactId>:<version>`. You can provide two or more KJAR files using the `|` separator, for example: `containerId=groupId:artifactId:version|c2=g2:a2:v2`. The Maven build process must produce all these files from the source in the Git repository.
```
Comment (sterobin) [inline doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: "The format is `<container...`" (no colon, per IBM, or make complete sentence)  Probably same at "separator, for example, `container...`. This would also be consistent with what you've done elsewhere in the doc.

File: doc-content/enterprise-only/openshift/operator-createapp-proc.adoc [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+*** `<url>`: The URL for the Git repository that contains the source for your decision service.
+*** `<branch>`: The branch in the Git repository.
+*** `<directory>`: The path to the source within the project downloaded from the Git repository.
```
Comment (sterobin) [inline doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: Not in this PR, but elsewhere in the doc, you use the format of listing the command and then "Where X is Y". You, Emily, and I discussed, and I think we agreed that, although IBM permits that, probably best to go with the form "Replace X with Y", like you've done in this section. Would be good to be consistent.

File: doc-content/enterprise-only/openshift/operator-createapp-proc.adoc [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+    roleFilter: (memberOf={1})
+    defaultRole: guest
+  roleMapper:
+    rolesProperties: /conf/roleMapper.properties
+    replaceRole: true
+----
++
+If the LDAP server does not define all the roles required for your deployment, you can map LDAP groups to {PRODUCT} roles. To enable LDAP role mapping, set the `rolesProperties` value to the fully qualified pathname of a file that defines role mapping, for example, `/opt/eap/standalone/configuration/rolemapping/rolemapping.properties`. You must provide this file and mount it at this path in all applicable deployment configurations; for instructions, see <<rolemapping-proc>>.
++
+If `replaceRole` is set to `true`, mapped roles replace the roles defined on the LDAP server; if it is set to `false`, both mapped roles and roles defined on the LDAP server are set as user application roles. The default setting is `false`.
```
Comment (sterobin) [inline doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: I think parameters and configs are not to stand alone if I recall, so would be something like "If the `replaceRole` parameter is...". Also, I'd say "server. If the parameter is set to" and nix the semicolon, or leave as worded but still no semicolon. You are a big fan of the semicolon I see :) Imo, it should be reserved for only very closely related clauses, where a comma isn't enough and a period is too much.

File: doc-content/enterprise-only/openshift/operator-createapp-proc.adoc [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+    roleAttributeID: memberOf
+    rolesCtxDN: ou=groups,ou=example,ou=com
+    roleFilter: (memberOf={1})
+    defaultRole: guest
+  roleMapper:
+    rolesProperties: /conf/roleMapper.properties
+    replaceRole: true
+----
++
+If the LDAP server does not define all the roles required for your deployment, you can map LDAP groups to {PRODUCT} roles. To enable LDAP role mapping, set the `rolesProperties` value to the fully qualified pathname of a file that defines role mapping, for example, `/opt/eap/standalone/configuration/rolemapping/rolemapping.properties`. You must provide this file and mount it at this path in all applicable deployment configurations; for instructions, see <<rolemapping-proc>>.
```
Comment (sterobin) [inline doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: "configurations. For instruction about LDAP role mapping [or whatever], see .."  (better/cleaner, imo)

File: doc-content/enterprise-only/openshift/operator-modify-proc.adoc [hunk 14]
```diff
@@ -0,0 +1,49 @@
+[id='operator-modify-proc']
+= Modifying an environment that is deployed using operators
+
+If an environment is deployed using operators, you can not modify it using typical OpenShift methods. For example, if you delete a pod, it is re-created automatically with the same parameters.
```
Comment (sterobin) [inline doc-content/enterprise-only/openshift/operator-modify-proc.adoc]: "you cannot"

File: doc-content/enterprise-only/openshift/operator-modify-proc.adoc [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+
+To modify the environment, you must modify the YAML description of the environment. You can change common settings, such as passwords, add new {KIE_SERVERS}, and scale {KIE_SERVERS}.
```
Comment (sterobin) [inline doc-content/enterprise-only/openshift/operator-modify-proc.adoc]: For clarity, I'd say "common settings such as passwords, add ..."

File: doc-content/enterprise-only/openshift/operator-createapp-proc.adoc [hunk 16]
```diff
@@ -0,0 +1,154 @@
...
++
+If `replaceRole` is set to `true`, mapped roles replace the roles defined on the LDAP server; if it is set to `false`, both mapped roles and roles defined on the LDAP server are set as user application roles. The default setting is `false`.
+. After completing the modification of the YAML source, click *Create* to create the application.
+
+[NOTE]
+====
+You can view other configuration samples, for example, for servers that use differen database servers, at https://github.com/kiegroup/kie-cloud-operator/tree/1.0.x/deploy/examples
```
Comment (sterobin) [inline doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: We generally don't point to outside repos, but in this case I agree because it's not something for our product, correct? Just samples? But either way, maybe truncate with linked text instead of the raw URL. Maybe "servers, see the https://...[operators samples repository] in GitHub." Or something.

File: doc-content/enterprise-only/openshift/operator-modify-proc.adoc [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+
+.Procedure
+. Enter your project in the OpenShift Web cluster console. 
+. In the OpenShift Web console navigation panel, select *Operators* and then *Subscriptions*.
+. Click the name of the subscription that containt `businessautomation`. Information about this subscription is displayed.
```
Comment (sterobin) [inline doc-content/enterprise-only/openshift/operator-modify-proc.adoc]: "contains"

File: doc-content/enterprise-only/openshift/operator-modify-proc.adoc [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+. Under the *Installed version* heading, click the version name of the subscription. An overview of the subscription is displayed.
+. Select the *Instances* tab.
+. Click the name of a deployed environment. 
+. Select the *YAML* tab. A YAML source is displayed.
+. If you want to change common settings, such as passwords, edit the values under `commonConfig:`.
+. If you want to add new {KIE_SERVERS}, add their descriptions at the end of the block under `servers:`. For example:
```
Comment (sterobin) [inline doc-content/enterprise-only/openshift/operator-modify-proc.adoc]: We supposedly should use colons with incomplete sentences, like "For example:", as we've discussed. Maybe "`servers:` as shown in the following example:" You use this frequently, and I actually don't disagree, just pointing it out so that you can choose to deliberately flout or address as needed.

Search all instances of "For/for example:" in the doc and address accordingly. Several instances.

File: doc-content/enterprise-only/openshift/operator-modify-proc.adoc [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+    kieServerContainerDeployment: <deployment>
+    gitSource:
+      uri: <url>
+      reference: <branch>
+      contextDir: <directory>
+----
++
+Replace the following values:
++
+*** `<deployment>`: The identifying information of the decision service (KJAR file) that is built from your source. The format is: `<containerId>=<groupId>:<artifactId>:<version>`. You can provide two or more KJAR files using the `|` separator, for example: `containerId=groupId:artifactId:version|c2=g2:a2:v2`. The Maven build process must produce all these files from the source in the Git repository.
```
Comment (sterobin) [inline doc-content/enterprise-only/openshift/operator-modify-proc.adoc]: Same comment as mine before, about "format is `<container...`" and "for example, `container...`"

File: doc-content/enterprise-only/openshift/operator-modify-proc.adoc [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+*** `<url>`: The URL for the Git repository that contains the source for your decision service.
+*** `<branch>`: The branch in the Git repository.
+*** `<directory>`: The path to the source within the project downloaded from the Git repository
```
Comment (sterobin) [inline doc-content/enterprise-only/openshift/operator-modify-proc.adoc]: "repository." (period, for consistency)

File: doc-content/enterprise-only/openshift/operator-modify-proc.adoc [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
++
+. If you want to scale a {KIE_SERVER}, find the description of the server in the block under `servers:`. Add a `replicas:` setting under that description; for example, `replicas: 3` scales the server to three pods.
```
Comment (sterobin) [inline doc-content/enterprise-only/openshift/operator-modify-proc.adoc]: I'd reword to "find the description ... and add a ... under that description. For example, `replicas: 3` ... ."

File: assemblies/assembly_openshift-operator/main.adoc [hunk 22]
```diff
@@ -0,0 +1,51 @@
+[id='assembly_openshift-operator']
+
+include::_artifacts/document-attributes.adoc[]
```
Comment (sterobin) [inline assemblies/assembly_openshift-operator/main.adoc]: Not in this PR, but the "Providing the LDAP role mapping file" section has a few things:

* " from the file. Run the following command:"  > "from the file by running the following command:" Right?
* "every deployment config ... . The following deployment configs ..."  >  "deployment configuration ... deployment configurations"
* Also, below the same section in my above bullet, the example snippet is missing. Goes from "can be affect in this environment: ..." to just "Where my-app ..."
* For every deployment configuration, run the command:"  >  "... run the following command:"

File: doc-content/enterprise-only/openshift/operator-createapp-proc.adoc [hunk 23]
```diff
@@ -0,0 +1,154 @@
+[id='operator-createapp-proc']
+= Deploying a {PRODUCT} environment using the operator
+
+After subscribing to the operator, you can _create an application_ to deploy an environment.
```
[the remark this answers is not in the document]
Comment (mramendi) [reply doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: Because it's a new term. It's not the vernacular meaning of "create an application".

File: doc-content/enterprise-only/openshift/operator-createapp-proc.adoc (hunk 16 above)
Comment (tchughesiv) [reply doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: @mramendi additionally as we prepare for 7.3.1, it's becoming apparent that we should use the release tag of `1.0.0` instead of `1.0.x` ... like so:
https://github.com/kiegroup/kie-cloud-operator/tree/1.0.0/deploy/examples

Can we change this everywhere it's referenced?

Comment (mramendi) [reply doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: In this case it's because it is tech preview. This SHOULD be downloadable from our Portal instead, but I'm not getting into this battle for this release.

File: doc-content/enterprise-only/openshift/operator-modify-proc.adoc (hunk 18 above)
Comment (mramendi) [reply doc-content/enterprise-only/openshift/operator-modify-proc.adoc]: Changed this instance, because of awkward punctuation. but not really up to handling all of the others now as they are non-trivial to change. most are not operat-ros-ecific and a full review of the template docs is planned for week after next.

File: doc-content/enterprise-only/openshift/operator-createapp-proc.adoc (hunk 16 above)
Comment (mramendi) [reply doc-content/enterprise-only/openshift/operator-createapp-proc.adoc]: @tchughesiv in the nick of time! changed just before a merge

**Comment by tchughesiv** [comment]
LGTM
